### PR TITLE
fix(api-v1): unblock REST SDK tasks stuck in RUNNING / task_busy

### LINKED ISSUE #739: bug(REST API): Xagent API tasks never reach “completed” status and always return `task_busy`

### PR DESCRIPTION
**Pull request by kejhz653-stack**
- Reset runner_id and lease columns in begin_turn atomic claim so stale leases from crashed workers do not block the bg runner on the next turn
- Stop execute_task from acquiring/releasing the task lease when called from execute_task_background; the orchestrator already owns that lifecycle and nested lease handling could leave rows terminal only in memory
- Retry lease acquisition once after clearing an expired stale holder
- Add orchestrator tests for lease reset and expired-lease recovery

Fixes xorbitsai/xagent#739

### CODE REVIEW

File: src/xagent/web/services/task_orchestrator.py [hunk 1]
```diff
@@ -510,6 +520,63 @@ def _refuse_if_bg_inflight(task_id: int) -> None:
         raise TaskTurnError("bg_inflight")
 
 
+def _retry_task_lease_acquire_after_stale_clear(task_id: int) -> Any:
```
Comment (rogercloud) [inline src/xagent/web/services/task_orchestrator.py]: MAJOR — Race condition (confirms the prior gemini-code-assist flag, with a concrete interleaving).

This does a read-then-write clear: read the task row, then an unconditional ORM `commit()` (UPDATE keyed only by primary key, no version column, no `SELECT ... FOR UPDATE`, no serializable isolation anywhere in this path) that nulls `runner_id`/`lease_expires_at`, before calling `acquire_task_lease` to reacquire.

Failure scenario: Worker A's first `acquire_task_lease_isolated` fails. A reads the row, sees `lease_expires_at` in the past, prepares to clear it. Between A's read and A's `commit()`, Worker C calls `acquire_task_lease` and atomically wins the (genuinely expired) lease, setting `runner_id=C` and a fresh future expiry. A's later unconditional `commit()` blindly overwrites C's columns back to NULL. A's subsequent `acquire_task_lease` call then also succeeds (row now shows `runner_id=NULL`). Both A and C now believe they hold the lease → double execution of the same task turn.

Also likely dead code for legitimate cases: `acquire_task_lease` only ever returns `None` when `lease_expires_at` is non-null and still in the future — in which case this function's own guard (`if expires_at is not None and expires_at >= utc_now(): return None`) declines to retry. So in every real production denial, this function does nothing; it only "fires" against inconsistent (test-mocked) state — see the CI-blocking regression this causes below.

Suggested fix: remove this function entirely (`acquire_task_lease`'s own atomic UPDATE already covers every case where a retry would legitimately help). If a manual clear must be kept, make it a single atomic conditional `UPDATE ... WHERE id=:id AND lease_expires_at < :now` with a rowcount check — never a blind ORM commit keyed only by primary key.

File: src/xagent/web/services/task_orchestrator.py [hunk 2]
```diff
@@ -510,6 +520,63 @@ def _refuse_if_bg_inflight(task_id: int) -> None:
...
+    from .task_lease_service import _rowcount, acquire_task_lease, utc_now
+
+    SessionLocal = get_session_local()
+    db = SessionLocal()
+    try:
+        task = db.query(Task).filter(Task.id == task_id).first()
+        if task is None:
+            return None
+        expires_at = task.lease_expires_at
+        if expires_at is not None and expires_at.tzinfo is None:
```
Comment (rogercloud) [inline src/xagent/web/services/task_orchestrator.py]: MINOR (nit) — Sets `last_heartbeat_at = utc_now()` while clearing the stale lease, but `begin_turn`'s equivalent reset (~L428-430) nulls `last_heartbeat_at` in the same situation. No functional impact confirmed (nothing reads `last_heartbeat_at` for liveness detection — `mark_task_paused_if_stale` keys only on `lease_expires_at`), and a subsequent successful `acquire_task_lease` overwrites it anyway. Consider nulling it to match `begin_turn`'s pattern for consistency — moot if this function is removed per the race-condition comment above.

File: src/xagent/web/api/chat.py [hunk 3]
```diff
@@ -1882,7 +1890,7 @@ async def execute_task(
         lease_heartbeat_task = None
         result: Dict[str, Any] | None = None
         sandbox_task_key = None
-        if db_session and tracker_task_id:
+        if manage_task_lease and db_session and tracker_task_id:
```
Comment (rogercloud) [inline src/xagent/web/api/chat.py:1893]: MINOR — `sync_workforce_run_status(..., TaskStatus.RUNNING)` (a few lines below, ~L1907) is nested inside this `manage_task_lease` condition, so it's now skipped whenever `manage_task_lease=False` — exactly the new bg path (`websocket.py:1381`). Verified: at base, `_schedule_bg`'s own lease acquisition never synced `RUNNING` (only `COMPLETED`/`FAILED`); this nested block was the only place syncing `RUNNING` for REST/SDK-triggered background tasks. Net effect: their workforce `TriggerRun` rows now stay `PENDING` for the whole run and jump straight to a terminal state, never transitioning through `RUNNING`. Likely cosmetic but real — consider gating the `RUNNING` sync on `db_session and tracker_task_id` alone (outside `manage_task_lease`), or adding an equivalent sync into `_schedule_bg`'s lease-acquire step.

File: tests/web/services/test_task_orchestrator.py [hunk 4]
```diff
@@ -857,6 +899,104 @@ async def test_schedule_bg_forwards_execution_message_to_execute_task_background
     assert kwargs["context"]["existing"] == "value"
 
 
+@pytest.mark.asyncio
+async def test_schedule_bg_acquires_after_expired_stale_lease(db_session) -> None:
+    """Expired lease columns must not permanently block the bg runner."""
+    from xagent.web.api.websocket import background_task_manager
+    from xagent.web.services.task_lease_service import utc_now
```
Comment (rogercloud) [inline tests/web/services/test_task_orchestrator.py:901]: MAJOR — This test does not exercise the retry function it claims to test.

`lease_expires_at` is set in the past, so the FIRST `acquire_task_lease_isolated` call already succeeds via `acquire_task_lease`'s own expired-lease WHERE clause — `_schedule_bg` never enters the `if lease is None:` branch that calls the new `_retry_task_lease_acquire_after_stale_clear`.

Verified empirically: patching `_retry_task_lease_acquire_after_stale_clear` to raise `AssertionError` and re-running this test — it still passes, proving the retry function is never invoked.

Suggested fix: construct a scenario where the first `acquire_task_lease_isolated` call genuinely fails (e.g. patch it to return `None` once, or set up a live lease held by a different `runner_id` that then expires) so the retry branch is actually exercised.

File: tests/web/services/test_task_orchestrator.py
Comment (rogercloud) [inline tests/web/services/test_task_orchestrator.py:1]: MAJOR (CI-blocking regression, pre-existing test not modified by this PR) — `test_schedule_bg_skips_finish_turn_when_lease_acquire_fails` (~L721) PASSES at base commit `87cead3` but FAILS at this PR's HEAD `c965be7` with `AssertionError: Expected mock to not have been awaited. Awaited 1 times.` (verified by checking out each revision and running pytest).

Root cause: this test mocks `acquire_task_lease_isolated` to return `None` (simulating "another worker holds the lease") but does not mock the new `_retry_task_lease_acquire_after_stale_clear` fallback that `_schedule_bg` now falls through to. The test's task row has default lease columns (`runner_id=None`, `lease_expires_at=None`) — inconsistent with the "held elsewhere" state the mock implies — so the retry function's guard doesn't trigger, and it proceeds to call the real `acquire_task_lease`, which succeeds against the actually-unheld row. The bg execution that this test asserts should be skipped now runs instead.

This is the same real-vs-mocked-state inconsistency that makes the retry function risky in production (see the race-condition comment on `_retry_task_lease_acquire_after_stale_clear` in task_orchestrator.py). Removing that function per that comment would make this test pass again unmodified; otherwise this test needs updating to also mock the fallback, e.g. `patch("xagent.web.services.task_orchestrator._retry_task_lease_acquire_after_stale_clear", return_value=None)`.

File: src/xagent/web/api/chat.py [hunk 5]
```diff
@@ -1857,6 +1857,8 @@ async def execute_task(
         task_id: Optional[str] = None,
         tracking_task_id: Optional[str] = None,
         db_session: Optional[Any] = None,
+        *,
```
Comment (rogercloud) [inline src/xagent/web/api/chat.py:1860]: MINOR — No test drives `execute_task` with a real `db_session` for either `manage_task_lease` branch. Tests exercising `_schedule_bg`/`execute_task_background` fully mock away `execute_task`, and tests calling the real `execute_task` (`tests/web/test_agent_manager_sandbox_lifecycle.py`) never pass a `db_session`, so the lease-handling branch is never reached either way. Suggest adding a focused unit test with `db_session` set — once with `manage_task_lease=True` (asserting acquire/release still happen, locking in the pre-existing default behavior) and once with `False` (asserting they're skipped, token tracking still runs, and `stop_task_lease_heartbeat(None, None)` in `finally` doesn't error).

**Review by rogercloud — CHANGES_REQUESTED** [review]
## Design Verdict: Acceptable with reservations

Two of the three changes are correctly targeted and fit the existing architecture:
- `begin_turn`'s atomic column reset (the root-cause fix for stale-lease-blocks-next-turn)
- `manage_task_lease=False` on the bg path (removing nested lease double-management between `execute_task` and the orchestrator)

The third piece — `_retry_task_lease_acquire_after_stale_clear` — is the problem: `acquire_task_lease` is already atomic and already grants a genuinely-expired lease via its own WHERE clause, so no separate manual clear step is needed. As written it (1) introduces a real read-then-write race, and (2) appears to be dead code for every legitimate production denial (its own guard clause declines to retry in exactly the states where `acquire_task_lease` actually returns `None`) — see inline comments.

**Design concern — duplication:** lease-clearing logic is now spread across three sites (`begin_turn`'s atomic UPDATE, this retry function's manual clear, and `mark_task_paused_if_stale`). Consolidating on the atomic-UPDATE pattern and removing the manual-clear variant is the cleaner path.

## Summary of findings (see inline comments for details)

**MAJOR**
1. Race condition in `_retry_task_lease_acquire_after_stale_clear` (confirms/extends the prior gemini-code-assist flag with a concrete interleaving) — `task_orchestrator.py`
2. New test `test_schedule_bg_acquires_after_expired_stale_lease` does not exercise the retry path it's meant to cover (empirically verified) — `test_task_orchestrator.py`
3. **CI-blocking regression, found by actually running the suite (not from the prior review):** the pre-existing, unmodified test `test_schedule_bg_skips_finish_turn_when_lease_acquire_fails` passes at base `87cead3` and fails at this PR's HEAD `c965be7` — `test_task_orchestrator.py`

**MINOR**
4. No test exercises `execute_task`'s new `manage_task_lease` parameter with a real `db_session` (either value) — `chat.py`
5. Workforce `TriggerRun` RUNNING-status sync is now skipped for REST/SDK background tasks (cosmetic regression) — `chat.py`
6. `last_heartbeat_at` handling in the stale-clear path is inconsistent with `begin_turn`'s equivalent reset (no functional impact) — `task_orchestrator.py`

Findings 1–3 point to the same root cause and the same fix: removing `_retry_task_lease_acquire_after_stale_clear` resolves the race, makes the uncovered test moot, and restores the broken pre-existing test without modifying it.

No issues found in: error handling, documentation, and API design (`manage_task_lease` keyword-only with a safe `True` default; verified no other `execute_task` call site needed the same treatment).

Requesting changes primarily due to Finding 3 (breaks CI) and Finding 1 (real double-execution race).

File: tests/web/services/test_task_orchestrator.py [hunk 6]
```diff
@@ -857,6 +894,58 @@ async def test_schedule_bg_forwards_execution_message_to_execute_task_background
     assert kwargs["context"]["existing"] == "value"
 
 
+@pytest.mark.asyncio
+async def test_schedule_bg_acquires_expired_lease_on_first_try(db_session) -> None:
+    """Expired lease columns are granted by acquire_task_lease's atomic WHERE."""
+    from xagent.web.api.websocket import background_task_manager
+    from xagent.web.services.task_lease_service import utc_now
+
+    user = _create_user(db_session)
```
Comment (rogercloud) [inline tests/web/services/test_task_orchestrator.py:903]: MINOR (residual, downgraded from the earlier MAJOR) — This test still doesn't route through `_schedule_bg`'s `if lease is None:` branch to reach `_retry_task_lease_acquire_after_stale_clear` end-to-end: `lease_expires_at` is set in the past, so the first `acquire_task_lease_isolated` call still succeeds directly via `acquire_task_lease`'s own expired-lease WHERE clause, and the retry function is never invoked here.

This is now lower severity since `_retry_task_lease_acquire_after_stale_clear` itself is directly unit-tested elsewhere (`test_retry_returns_none_when_lease_is_live`, `test_retry_stale_clear_clears_expired_lease_and_acquires`), but there's still no integration-level test proving `_schedule_bg` correctly calls the retry function and consumes its return value when the *first* `acquire_task_lease_isolated` call genuinely fails. Suggest either renaming this test to reflect what it actually covers (the pre-existing expired-lease disjunct), or adding a companion test that mocks `acquire_task_lease_isolated` to return `None` once to force the real end-to-end retry path.

File: src/xagent/web/api/chat.py [hunk 7]
```diff
@@ -1882,14 +1890,20 @@ async def execute_task(
         lease_heartbeat_task = None
         result: Dict[str, Any] | None = None
         sandbox_task_key = None
-        if db_session and tracker_task_id:
+        if manage_task_lease and db_session and tracker_task_id:
```
Comment (rogercloud) [inline src/xagent/web/api/chat.py:1893]: Not fixed — still open from the prior review round. `sync_workforce_run_status(..., TaskStatus.RUNNING)` (a few lines below, ~1907) remains nested inside this `manage_task_lease` condition, so it's still skipped whenever `manage_task_lease=False` (the bg path). Verified `chat.py` has zero diff between the previously-reviewed commit and this one. Net effect unchanged: REST/SDK-triggered background tasks' workforce `TriggerRun` rows still stay `PENDING` for the whole run and jump straight to a terminal state, never transitioning through `RUNNING`. Suggest gating the `RUNNING` sync on `db_session and tracker_task_id` alone (outside `manage_task_lease`), or adding an equivalent sync into `_schedule_bg`'s lease-acquire step.

Comment (rogercloud) [inline src/xagent/web/api/chat.py:1893]: Not fixed — still open from the prior review round. `sync_workforce_run_status(..., TaskStatus.RUNNING)` (a few lines below, ~1907) remains nested inside this `manage_task_lease` condition, so it's still skipped whenever `manage_task_lease=False` (the bg path). Verified `chat.py` has zero diff between the previously-reviewed commit and this one. Net effect unchanged: REST/SDK-triggered background tasks' workforce `TriggerRun` rows still stay `PENDING` for the whole run and jump straight to a terminal state, never transitioning through `RUNNING`. Suggest gating the `RUNNING` sync on `db_session and tracker_task_id` alone (outside `manage_task_lease`), or adding an equivalent sync into `_schedule_bg`'s lease-acquire step.

File: src/xagent/web/api/chat.py (hunk 5 above)
Comment (rogercloud) [inline src/xagent/web/api/chat.py:1860]: Not fixed — still open from the prior review round. `grep -rn "manage_task_lease" tests/` still returns zero hits: no test drives `execute_task` with a real `db_session` for either `manage_task_lease` branch. This commit's diff didn't touch `chat.py`, so the gap remains exactly as before. Suggest adding a focused unit test with `db_session` set — once with `manage_task_lease=True` (locking in the pre-existing default behavior) and once with `False` (asserting lease acquire/release/heartbeat are skipped, token tracking still runs, `stop_task_lease_heartbeat(None, None)` doesn't error).

File: src/xagent/web/services/task_orchestrator.py
Comment (rogercloud) [inline src/xagent/web/services/task_orchestrator.py:1]: MINOR (residual, not fully covered) — in `_retry_task_lease_acquire_after_stale_clear` (~L567), the CAS `rowcount != 1` branch (the "another worker changed the row between read and clear" race-detected path) is still not exercised by any test. The two new tests (`test_retry_returns_none_when_lease_is_live`, `test_retry_stale_clear_clears_expired_lease_and_acquires`) cover the early-return and the successful-clear paths, but not this one. Reliably reproducing the exact read/CAS interleaving in a unit test needs a hook injected between the read and the UPDATE (e.g. monkeypatching `db.execute` to mutate the row first) — worth adding since this is the one branch that exists specifically to handle the race this function was rewritten to fix.

Not fixed (nit, cosmetic) — ~L562 still sets `last_heartbeat_at=utc_now()` while clearing the stale holder, whereas `begin_turn`'s equivalent reset (~L428-430) nulls `last_heartbeat_at` in the same situation. Confirmed no functional impact (nothing in the codebase reads `last_heartbeat_at` for staleness/liveness detection — `mark_task_paused_if_stale` keys only on `lease_expires_at`), and a subsequent successful `acquire_task_lease` overwrites it anyway. Purely a readability/consistency nit; feel free to fold into a follow-up rather than blocking on it.

**Review by rogercloud — COMMENTED** [review]
## Re-review of `a6d30356` (fixes for the prior REQUEST_CHANGES round)

Not approving yet — flagging what's still open below. The two truly blocking issues from the prior round are resolved:

**Fixed & verified:**
- The race condition in `_retry_task_lease_acquire_after_stale_clear` is fixed — the clear is now a compare-and-swap conditional UPDATE keyed on a snapshot of `runner_id`/`lease_expires_at`, with a `rowcount` check before falling through to `acquire_task_lease`. Replayed the original interleaving: if another worker wins the lease mid-function, the CAS now matches 0 rows and does not clobber it.
- The CI-blocking regression (`test_schedule_bg_skips_finish_turn_when_lease_acquire_fails`) is fixed — the test now also mocks the retry fallback, and the full file passes 31/31.

**Not fully fixed / still open (see inline comments):**
- The new retry-function tests are real (run against an actual SQLite engine, no mocking) and cover the early-return and successful-clear paths, but two residual gaps remain: the CAS `rowcount != 1` race-detected branch is untested, and the original `test_schedule_bg_acquires_after_expired_stale_lease` still doesn't route through `_schedule_bg`'s retry branch end-to-end.
- Two MINOR findings from the prior round are untouched, since this commit only modified `task_orchestrator.py`/its tests and did not touch `chat.py`: no test drives `execute_task`'s `manage_task_lease` parameter with a real `db_session`, and the workforce `TriggerRun` RUNNING-status sync is still skipped for `manage_task_lease=False` background tasks.
- One cosmetic nit (`last_heartbeat_at=utc_now()` vs `begin_turn`'s `None` pattern in the stale-clear path) is also still open — no functional impact, low priority.

None of the open items are blocking on their own, but leaving them as explicit comments rather than approving so they don't get lost.

File: tests/web/test_execute_task_manage_task_lease.py [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+            task="hello",
+            tracking_task_id=str(task.id),
+            db_session=db_session,
+            manage_task_lease=False,
+        )
+
+    assert result["success"] is True
+    mock_acquire.assert_not_called()
+    mock_release.assert_not_called()
+    mock_sync.assert_called_once()
```
Comment (rogercloud) [inline tests/web/test_execute_task_manage_task_lease.py:152]: Non-blocking follow-up suggestion: neither test in this file asserts that `sync_workforce_run_status` is a no-op (no duplicate write/commit) when the `WorkforceRun` is already `running`. Since this PR makes the sync call unconditional on `manage_task_lease`, idempotency across repeat invocations (e.g. multiple turns on the same task) is the key safety assumption behind the fix — worth a small test asserting a second call with an already-`running` run doesn't re-commit.

**Review by rogercloud — APPROVED** [review]
## Re-review of `1c757d2a` — Approve

This follow-up commit correctly addresses everything flagged in the prior COMMENT round:

- `sync_workforce_run_status` now runs unconditionally of `manage_task_lease` (moved out of the lease-acquire block in `chat.py`), fixing the missed workforce RUNNING sync for the bg-runner (`manage_task_lease=False`) path.
- New `tests/web/test_execute_task_manage_task_lease.py` covers `AgentServiceManager.execute_task` against a real `db_session` for both `manage_task_lease=True` and `manage_task_lease=False`, closing the previous coverage gap.
- `_retry_task_lease_acquire_after_stale_clear` was removed entirely rather than patched further. Verified this is safe, not a regression: `acquire_task_lease`'s own atomic `WHERE lease_expires_at < now` clause already grants everything the retry covered, and `_begin_turn_atomic_sync` clears a stale/crashed lease on the next turn claim. No scenario the retry protected against is left unrecovered.
- Full relevant test suite (`test_task_orchestrator.py`, `test_execute_task_manage_task_lease.py`, `test_websocket_uploaded_files_context.py`) passes: 60+ tests, no regressions.

One non-blocking suggestion left as an inline comment: add a test asserting `sync_workforce_run_status` is idempotent (no duplicate commit) when the run is already `running`, since that's the key assumption behind making the call unconditional now.

Not blocking merge — approving.

File: src/xagent/web/services/task_orchestrator.py [hunk 9]
```diff
@@ -418,6 +418,16 @@ def _begin_turn_atomic_sync(
...
+                    # A new turn owns the lifecycle from this claim forward.
+                    # Stale runner_id / lease columns left by a crashed worker
+                    # or a release that failed to clear would otherwise make
+                    # acquire_task_lease deny this worker even though no live
+                    # runner is executing -- leaving the SDK row stuck RUNNING
+                    # (GET /v1/chat/tasks/{id} never reaches completed; append
+                    # returns task_busy). Reset here atomically with the claim.
+                    Task.runner_id: None,
+                    Task.lease_expires_at: None,
+                    Task.last_heartbeat_at: None,
```
Comment (rogercloud) [inline src/xagent/web/services/task_orchestrator.py:430]: Non-blocking follow-up (pre-existing behavior, not introduced by this PR): this reset now also clears `lease_expires_at` on the crash-recovery/APPEND path (previously it was already `None` for fresh PENDING tasks, but a crashed run's stale, non-None `lease_expires_at` is now nulled here too). Between this claim committing and the background runner actually calling `acquire_task_lease` (later, async), there's a brief window where the row is `RUNNING` with `lease_expires_at=None`.

`mark_task_paused_if_stale` (`src/xagent/web/services/task_lease_service.py:219`) treats `lease_expires_at is None` as stale rather than skipping it, and is reachable from concurrent status-poll paths (`chat.py`'s `GET /task/{task_id}`, `websocket.py`'s historical-data stream). So a poll landing in that window could flip a just-claimed RUNNING task to PAUSED/FAILED.

This self-corrects — the legitimate worker's subsequent `acquire_task_lease` call matches on `status != RUNNING` and restores RUNNING — so it's not task-loss in the common case, and the exposure already existed for fresh PENDING tasks. But this change does widen it slightly to the crash-recovery path. Worth considering a short grace-period guard in `mark_task_paused_if_stale` (e.g. skip the stale check if `task.updated_at` is very recent, or don't treat `lease_expires_at is None` as stale immediately after a claim).

File: tests/web/services/test_task_orchestrator.py [hunk 10]
```diff
@@ -162,6 +162,43 @@ async def test_begin_turn_create_clears_no_terminal_fields_when_pending(
     assert task.input == "first turn"
     assert task.output is None
     assert task.error_message is None
+    assert task.runner_id is None
+    assert task.lease_expires_at is None
```
Comment (rogercloud) [inline tests/web/services/test_task_orchestrator.py:166]: Minor: these two new assertions (`task.runner_id is None`, `task.lease_expires_at is None`) aren't actually a regression guard for the new reset logic in `_begin_turn_atomic_sync` — `_create_task` never sets `runner_id`/`lease_expires_at`, so a freshly created PENDING task already has both as `None` before `begin_turn` runs. They'd pass identically with or without the reset code.

Not a blocker — `test_begin_turn_claim_clears_stale_lease_columns` right below this one does properly guard the reset (it sets a real stale `runner_id`/future `lease_expires_at` before the claim), so the APPEND/crash-recovery path is covered. Just flagging that this specific pair of assertions documents intent rather than testing behavior.

**Review by rogercloud — COMMENTED** [review]
Re-reviewed the latest push (rebased onto main; prior 5 commits squashed into `a370365b`, content-verified identical to the previously-approved round, plus one genuinely new fix in the same commit and one new commit on top).

New since the last approval:
- `_begin_turn_atomic_sync` (task_orchestrator.py) now also resets `runner_id`/`lease_expires_at`/`last_heartbeat_at` to `None` atomically on every turn claim, fixing #739 (REST/SDK tasks stuck in RUNNING/task_busy due to stale lease columns from a crashed worker blocking the subsequent `acquire_task_lease`). Design is sound — the reset only fires on the atomic claim's WHERE clause (PENDING for CREATE, COMPLETED/FAILED/PAUSED for APPEND), which structurally excludes RUNNING rows, so a live worker's lease can't be clobbered.
- A new test asserting `sync_workforce_run_status` is idempotent when the run is already `running` — addresses the test-coverage gap from the last round.

Two non-blocking follow-ups left as inline comments (both low-severity, neither introduced by this PR nor blocking):
1. A pre-existing gap in `mark_task_paused_if_stale` (no guard for `lease_expires_at is None`) is slightly widened by the new reset — self-corrects via `acquire_task_lease`, so no task loss in the common case, but worth a small defensive fix.
2. One of the two new orchestrator tests has assertions that pass regardless of the fix (vacuous for the CREATE path) — the sibling test for the APPEND/crash-recovery path is a genuine regression guard, so coverage is fine overall, just noting the redundant assertions.

Full relevant test suite (`test_task_orchestrator.py` + `test_execute_task_manage_task_lease.py` + `test_websocket_uploaded_files_context.py`) passes with no regressions. Not blocking — prior APPROVE stands.
